### PR TITLE
Admin can only create categories/sections preliminary

### LINKED ISSUE #24: Only admins should be able to create sections
This was moved from #16 to make it not blocked.

### PR DESCRIPTION
**Pull request by enico**
fix #24 not ready for merge probably will change depending on #52, and could be a better way on doing this.  No tests have been written yet either. Comments, suggestions?

### CODE REVIEW

**Comment by enico** [comment]
okay that definitely failed

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -2,4 +2,11 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+  
+  def confirm_admin
+    unless current_user.admin?
+      flash[:alert] = "Test not Admin"
```
Comment (kenmazaika) [inline app/controllers/application_controller.rb]: We may want to change the message.  What do you think about: 

> User doesn't have required permissions

Or something like that?

File: app/controllers/sections_controller.rb [hunk 2]
```diff
@@ -1,5 +1,6 @@
 class SectionsController < ApplicationController
-  before_action :authenticate_user!, only: [:new, :create]
+  #before_action :authenticate_user!, only: [:new, :create]
```
Comment (kenmazaika) [inline app/controllers/sections_controller.rb]: I think we probably still want to have the `before_action :authenticate_user!`

**Comment by kenmazaika** [comment]
I made a couple small nitpicks.  It also looks like our tests now are failing, so I think we may need to fix that before we merge this too.  Overall it looks solid though!

**Comment by enico** [comment]
So, I'm not entirely sure about the rearrangement of tests, it was acting wonky with the confirm_admin helper. So I kind of imagined everything from an admin trying to access, and rejecting if not. But if you're not logged in, it does not redirect to a login, just to the confirm admin helper(root). Thoughts?

**Comment by kenmazaika** [comment]
Tests look good to me.  I'm happy to merge this if you want me to (I think it's ready).  Let me know if I should hit the button!

**Comment by enico** [comment]
yep good to go

**Comment by kenmazaika** [comment]
:+1:

**Pull request merged**